src/TypeSpec.Extension/Emitter.Csharp/src/lib/operation.ts

```diff
@@ -116,19 +112,28 @@ export function loadOperation(
             typespecParameters.body.type
         );
         if (effectiveBodyType.kind === "Model") {
-            if (effectiveBodyType.name !== "") {
-                parameters.push(
-                    loadBodyParameter(sdkContext, effectiveBodyType)
-                );
-            } else {
-                effectiveBodyType.name = `${capitalize(op.name)}Request`;
-                let bodyParameter = loadBodyParameter(
-                    sdkContext,
-                    effectiveBodyType
-                );
+            let bodyParameter = loadBodyParameter(
+                sdkContext,
+                effectiveBodyType
+            );
+            if (effectiveBodyType.name == "") {
                 bodyParameter.Kind = InputOperationParameterKind.Spread;
-                parameters.push(bodyParameter);
             }
+            // TODO: remove this after https://github.com/Azure/typespec-azure/issues/69 is resolved
+            // workaround for alias model
+            if (bodyParameter.Type.Name == "") {
+                // give body type a name
+                bodyParameter.Type.Name = `${capitalize(op.name)}Request`;
+                var bodyModelType = bodyParameter.Type as InputModelType;
+                bodyModelType.Usage = Usage.Input;
+                // update models cache
+                models.delete("");
```

Comment:
It may not good add model first, then remove it. Better to adjust the code structure to add once.

Reply:
Agree. But this is just a work-around, so I tend to concentrate in one place, so that later on it's easy to remove them all.